wondrous-app/components/Common/Image/SafeImage.tsx

```diff
@@ -66,7 +67,7 @@ export function SafeImage(safeImageArgs: SafeImageArgs) {
     },
   });
 
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
Is there a reason we're using this hook, isn't it blocking?

Reply:
I think it make sense to have `useLayoutEffect` here because in case we have image cached it will be rendered without "blinking" because `useEffect` will be triggered after showing image in DOM with placeholder. But if it's not cached - nothing changes

Reply:
but yea, it will be run synchronously

Reply:
this is my only concern too, tbh not super familiar @vefimov do you have any thoughts?

Reply:
I suppose this still happens after dom render

Reply:
Yes, it will run after DOM mutations but before browser rendering, so you won't see any visual blinks. But it runs synchronously and will block rendering and in our case it won't affect anything, because there are not many calculations

Reply:
I think it's okay to use useLayoufEffect here. We'll avoid flickering if the img is cached

Reply:
preferably use `useEffect` to avoid blocking visual updates + we've to call `getPreviewFile` as soon as the component is mounted.  

@lisovskyR can you show an example of when the image is blinking?

Reply:
Sure, here I rolled back to `useEffect`, look at top-left org icon and card assignee image:
https://user-images.githubusercontent.com/37869104/203975249-4b380893-664b-4b17-b95e-4b6566d19c65.mov

with `useLayoutEffect` there is no such thing

Reply:
I mean, this can happen in different scenarios because 1st render will be without any image

Reply:
Thank you. I see flashing when moving the card to another column even with `useLayoutEffect`. The root cause is related to the re-render. Please look on the video below:
https://www.loom.com/share/9d9fdf4341104b44b53cae58c9d04a9a

Reply:
And i think we can add memoization for image component to avoid it's re-render everywhere

Reply:
<img width="630" alt="Screenshot 2022-11-25 at 18 38 56" src="https://user-images.githubusercontent.com/37869104/204017917-b7cf3fbf-b37b-4e52-bfcc-12012b1360db.png">
Right now OrgProfilePicture not only rerendered many times, it even get unmounted every time

Reply:
> <img alt="Screenshot 2022-11-25 at 18 38 56" width="630" src="https://user-images.githubusercontent.com/37869104/204017917-b7cf3fbf-b37b-4e52-bfcc-12012b1360db.png">
> 
> Right now OrgProfilePicture not only rerendered many times, it even get unmounted every time

yeah :-(

Reply:
There are actually two problems with sidebar:
- EntitySidebar creates Sidebar every time it got re-rendered
- OrgProfilePicture has inline styles for SafeImage so it will be a new object every time and memoization in SafeImage won't help

Reply:
i think it's better to fix this in separate pr, because it require separate testing and review

Reply:
Ok, so I've added a fix for EntitySidebar memoization, now it won't be re-rendered

Reply:
I've fixed un-mounts in my upcoming PR, so don't worry about it atm

Reply:
But anyway, it's good to have more memoization here